File: gcc/testsuite/gcc.dg/loop-7.c
/* PR rtl-optimization/31360  */

/* { dg-do compile { target { powerpc*-*-* } } } */
/* { dg-options "-O1 -fdump-rtl-loop2_invariant" } */

void f(int *a)
{
  int i;
  for (i = 0;i<100;i++)
    a[i] = 0;
}

/* Load of 0 is moved out of the loop.  */
/* { dg-final { scan-rtl-dump-times "Decided" 1 "loop2_invariant" } } */

File: gcc/testsuite/rust/compile/torture/loop7.rs
fn main() {
    let mut a = 1;
    let mut b = 1;

    let _fib = loop {
        if (a % 2 == 0) {
            continue;
        }
        let c = a + b;
        a = b;
        b = c;
    };
}
